apps/frontend/components/Report/Report.tsx

```diff
@@ -0,0 +1,23 @@
+import {
+  DroughtDto,
+  FloodDto,
+  IncidentDto,
+  isDroughtArray,
+  isFloodArray,
+} from '@wfp-dmp/interfaces';
+
+import { FloodReport } from './FloodReport/FloodReport';
+
+export const Report = ({
+  forms,
+}: {
+  forms: FloodDto[] | DroughtDto[] | IncidentDto[];
+}) => {
+  if (forms.length === 0) return null;
+
+  if (isFloodArray(forms)) return <FloodReport forms={forms} />;
```

Comment:
Don't we know which type of report has been selected using the checkbox?

Reply:
True, here the type guard helps me for the typing too